app/serializers/post_revision_serializer.rb

```diff
@@ -279,7 +279,7 @@ def hidden_tags
   end
 
   def filter_tags(tags)
-    tags - hidden_tags
+    tags.is_a?(Array) && tags.any? ? tags - hidden_tags : tags
```

Comment:
Could a hidden tag passed as a string slip by, though? Super edge case...

Reply:
Possibly, but unless we do an `eval tags` there's no easy way to know 😬 

TBF, I think the serialization issue only happens on empty arrays.

Reply:
Plus, that behavior was introduced about ~6 years ago and hasn't changed since I refactored that part a few weeks ago and didn't know about that edge case.

Reply:
Ok!

> TBF, I think the serialization issue only happens on empty arrays.

Could it be a tiny bit safer to check specifically for an empty `tags`? Your call, PR approved.

Reply:
Not sure I understand what you mean. When `tags` is a string, what do you want us to do?

Reply:
My bad, when `tags` is a string, it is not necessarily empty. 

Maybe the right approach is to treat it as an empty array if it's a string. I.e. 

```suggestion
    tags.is_a?(Array) && tags.any? ? tags - hidden_tags : []
```

This way there is no chance hidden tags will leak, and when comparing previous to current in `tags_changes`, it will still be comparing arrays to arrays (vs passing the string as is).

Reply:
🤔 I need to check whether my assumption about it being a string when it's empty is correct. Will revisit tomorrow.

Thanks